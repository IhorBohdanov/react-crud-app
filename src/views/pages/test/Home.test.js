import React from "react";
import { render, screen } from "@testing-library/react";
import Home from "../Home";


describe("Home page", () => {
  beforeEach(() => {
    render(<Home />);
  });

  it('should match snapshot', () => {
    const { asFragment } = render(<Home />);
    expect(asFragment(<Home />)).toMatchSnapshot();
  });

  it("should render heading", () => {
    const heading = screen.getByRole("heading");
    expect(heading).toBeInTheDocument();
    expect(heading).toHaveTextContent(/Home page/i);
  });

  it("should render requirements list", () => {
    const list = screen.getByRole('list');
    const listItems = screen.getAllByRole('listitem');
    expect(list).toBeInTheDocument();
    expect(listItems).toHaveLength(5);
  })
});
