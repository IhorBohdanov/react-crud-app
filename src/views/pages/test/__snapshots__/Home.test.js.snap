// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page match snapshot 1`] = `
<DocumentFragment>
  <h1>
    Home page
  </h1>
  <ol>
    <li>
      Реализовать CRUD Web приложение используя React
    </li>
    <li>
      В качестве бэкэнда использовать сервис http://jsonplaceholder.typicode.com/
    </li>
    <li>
      Вывести список альбомов используя API GET http://jsonplaceholder.typicode.com/albums
    </li>
    <li>
      Вывести список фото для каждого альбома, используя роутинг и API GET http://jsonplaceholder.typicode.com/albums/1/photos где 1 - id альбома.
    </li>
    <li>
      Реазизовать добавление/изменение/удаление альбомов, используя API POST/PUT/DELETE http://jsonplaceholder.typicode.com/albums/1
    </li>
  </ol>
</DocumentFragment>
`;

exports[`Home page should match snapshot 1`] = `
<DocumentFragment>
  <h1>
    Home page
  </h1>
  <ol>
    <li>
      Реализовать CRUD Web приложение используя React
    </li>
    <li>
      В качестве бэкэнда использовать сервис http://jsonplaceholder.typicode.com/
    </li>
    <li>
      Вывести список альбомов используя API GET http://jsonplaceholder.typicode.com/albums
    </li>
    <li>
      Вывести список фото для каждого альбома, используя роутинг и API GET http://jsonplaceholder.typicode.com/albums/1/photos где 1 - id альбома.
    </li>
    <li>
      Реазизовать добавление/изменение/удаление альбомов, используя API POST/PUT/DELETE http://jsonplaceholder.typicode.com/albums/1
    </li>
  </ol>
</DocumentFragment>
`;
